### PR TITLE
[AKS] Change aks create to enable managed identity by default

### PR DESCRIPTION
**Pull request by norshtein**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

 This PR changes command `az aks create` to enable managed identity for the cluster by default.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by norshtein** [comment]
@qwordy  @zqingqing1 @feiskyer could you please have a review?

**Comment by norshtein** [comment]
Close and reopen to rerun CI.

**Comment by yonzhan** [comment]
add to S166.

**Comment by andyzhangx** [comment]
/hold
hold this PR since whether make msi by default in aks cli should be discussed first

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -1815,6 +1816,15 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
     if all([disable_rbac, enable_rbac]):
         raise CLIError('specify either "--disable-rbac" or "--enable-rbac", not both.')
 
+    identity = None
+    # Enable managed identity by default if customer
+    # doesn't explicity specify a service principal in
+    # the request.
+    if not service_principal:
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py]: default MSI on?

**Comment by andyzhangx** [comment]
cc @sauryadas

**Comment by norshtein** [comment]
Closed. In our internal discussion, we decide not to enable managed identity by default, will use a flag instead.

**Pull request closed without merging**